langchain/chat_models/base.py

```diff
@@ -0,0 +1,49 @@
+from abc import ABC, abstractmethod
+from typing import List, Optional
+
+from langchain.schema import BaseMessage, ChatGeneration, ChatResult, LLMResult
+
+
+class BaseChatModel(ABC):
+    def generate(
+        self, messages: List[List[BaseMessage]], stop: Optional[List[str]] = None
+    ) -> LLMResult:
+        """Top Level call"""
+        results = []
+        for m in messages:
+            results.append(self._generate(m, stop=stop))
+        return LLMResult(generations=[res.generations for res in results])
```

Comment:
Am I missing something here? I don't think this should work. `LLMResult` expects a `List[List[Generation]]`
and `ChatResult` has a `List[ChatGeneration]`, we need to either allow both `ChatGeneration` or `Generation` in `LLMResult` or do something else